app/src/test/java/org/dhis2/data/filter/FilterRepositoryTest.kt

```diff
@@ -85,4 +105,269 @@ class FilterRepositoryTest {
+        val program = Program.builder().uid("random")
+            .programType(org.hisp.dhis.android.core.program.ProgramType.WITH_REGISTRATION).build()
+        whenever(d2.settingModule().appearanceSettings().blockingExists()) doReturn true
+        whenever(
+            d2.settingModule().appearanceSettings().getProgramFiltersByUid(program.uid())
+        ) doReturn emptyMap()
+        whenever(
+            getFiltersApplyingWebAppConfig.execute(
+                any<LinkedHashMap<ProgramFilter, FilterItem>>(),
+                any<Map<ProgramFilter, FilterSetting>>()
+            )
+        ) doReturn emptyList()
+
+        val result = filterRepository.dashboardFilters(program.uid())
+
+        assert(result.isEmpty())
+    }
+
+    @Test
+    fun `Should get tracker filters with working list when webapp is configured`() {
+        val program = Program.builder().uid("random")
+            .programType(org.hisp.dhis.android.core.program.ProgramType.WITH_REGISTRATION).build()
+        whenever(d2.programModule().programs().uid(any())) doReturn mock()
+        whenever(d2.programModule().programs().uid(any()).get()) doReturn Single.just(program)
+        whenever(d2.settingModule().appearanceSettings().blockingExists()) doReturn true
+        whenever(
+            d2.programModule().programStages().byProgramUid().eq(program.uid())
+        ) doReturn mock()
+        whenever(
+            d2.programModule().programStages().byProgramUid().eq(program.uid())
+                .byEnableUserAssignment()
+        ) doReturn mock()
+        whenever(
+            d2.programModule().programStages().byProgramUid().eq(program.uid())
+                .byEnableUserAssignment().eq(true)
+        ) doReturn mock()
+        whenever(
+            d2.programModule().programStages().byProgramUid().eq(program.uid())
+                .byEnableUserAssignment().eq(true).blockingIsEmpty()
+        ) doReturn false
+        whenever(
+            d2.trackedEntityModule().trackedEntityInstanceFilters().byProgram().eq(program.uid())
+        ) doReturn mock()
+        whenever(
+            d2.trackedEntityModule().trackedEntityInstanceFilters().byProgram().eq(program.uid())
+                .withTrackedEntityInstanceEventFilters()
+        ) doReturn mock()
+        whenever(
+            d2.trackedEntityModule().trackedEntityInstanceFilters().byProgram().eq(program.uid())
+                .withTrackedEntityInstanceEventFilters().blockingGet()
+        ) doReturn createTrackerEntityInstanceFilters()
+        whenever(
+            d2.settingModule().appearanceSettings().getProgramFiltersByUid(program.uid())
+        ) doReturn createWebAppTrackedEntityFilters()
+        whenever(
+            getFiltersApplyingWebAppConfig.execute(
+                any<LinkedHashMap<ProgramFilter, FilterItem>>(),
+                any<Map<ProgramFilter, FilterSetting>>()
+            )
+        ) doReturn createGlobalTrackerFilterResult()
```

Comment:
I'm cleaning this up, but yes we really need to work on a proposal for the SDK in order to reduce this boilerplate